sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -80,16 +87,80 @@ internal static string GenerateUserAgentString(Assembly clientAssembly, string?
             {
                 version = version.Substring(0, hashSeparator);
             }
-            var platformInformation = $"({RuntimeInformation.FrameworkDescription}; {RuntimeInformation.OSDescription})";
+            runtimeInformation ??= new RuntimeInformationWrapper();
+            var platformInformation = EscapeProductInformation($"({runtimeInformation.FrameworkDescription}; {runtimeInformation.OSDescription})");
 
-            return applicationId != null
+            var ua = applicationId != null
                 ? $"{applicationId} azsdk-net-{assemblyName}/{version} {platformInformation}"
                 : $"azsdk-net-{assemblyName}/{version} {platformInformation}";
+            return ua;
         }
 
         /// <summary>
         /// The properly formatted UserAgent string based on this <see cref="TelemetryDetails"/> instance.
         /// </summary>
         public override string ToString() => _userAgent;
+
+        /// <summary>
+        /// If the ProductInformation is not in the proper format, this escapes any ')' , '(' or '\' characters per https://www.rfc-editor.org/rfc/rfc7230#section-3.2.6
+        /// </summary>
+        /// <param name="productInfo">The ProductInfo portion of the UserAgent</param>
+        /// <returns></returns>
+        private static string EscapeProductInformation(string productInfo)
+        {
+            // If the string is already valid, we don't need to escape anything
+            bool success = false;
+            try
+            {
+                success = ProductInfoHeaderValue.TryParse(productInfo, out var _);
+            }
+            catch (Exception)
```

Comment:
Would it make sense to scope this to specific exception types?

Reply:
In this case, I don't think it does. In all cases we want to treat it as a failure to parse correctly.